lib/components/Upload.js

```diff
@@ -293,6 +343,28 @@ export default class Upload extends Component {
     );
   }
 
+  renderMedtronicSerialNumberInput() {
+    const { upload } = this.props;
+    if (_.get(upload, 'source.driverId', null) !== 'Medtronic') {
+      return null;
+    }
+
+    return (
+      <div>
+        <div className={styles.textInputWrapper}>
+          <p>Enter your 6 digit serial number found on the back of your pump.</p>
+          <input
+            type="number"
```

Comment:
So tricky! `input type="number"` doesn't emit an `onChange` event when the input is `123e` until something comes AFTER the `e` (like `123e4`), so it was still allowing the `e` to be entered. I switched the `input` type to `text`, to resolve that.